### PR TITLE
PR - Neato plugin

### PR DESCRIPTION
**Pull request by thengsty**

### CODE REVIEW

**Comment by psilo909** [comment]
i am missing the requirements.txt file. from what i see, at least requests is required as library?

**Comment by thengsty** [comment]
isn't it part of shng core? In this case I am going to remove the hint from README

**Comment by psilo909** [comment]
each plugin should bring its own requirements, even if they are also somewhere else.. e.g. the version of the requirement might differ and we show the requirements in the backend / admin interface and list the plugins which have them

**Comment by thengsty** [comment]
OK, thanks for clarification. Added requirements.txt

File: neato/__init__.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    _items = []
+
+    def __init__(self, sh, *args, **kwargs):
+        from bin.smarthome import VERSION
+        if '.'.join(VERSION.split('.', 2)[:2]) <= '1.5':
+            self.logger = logging.getLogger(__name__)
+
+        self.robot = Robot(self.get_parameter_value("account_email"), self.get_parameter_value("account_pass"))
+        self.robot.update_robot()
+        self._cycle = 20
```
[the remark this answers is not in the document]
Comment (thengsty) [reply neato/__init__.py]: changed to 40 secs. But should not be more to have an always updated robot in SHNG

File: neato/plugin.yaml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+#    tester:                         # Thomas Hengsberg
+    state: develop                  # change to ready when done with development
+#    keywords: iot xyz
+    documentation: https://github.com/smarthomeNG/smarthome/wiki/CLI-Plugin        # url of documentation (wiki) page
+#    support: https://knx-user-forum.de/forum/supportforen/smarthome-py
+
+    version: 1.6.0                  # Plugin version
+    sh_minversion: 1.6              # minimum shNG version to use this plugin
+#    sh_maxversion:                 # maximum shNG version to use this plugin (leave empty if latest)
+    multi_instance: False           # plugin supports multi instance
```
[the remark this answers is not in the document]
Comment (thengsty) [reply neato/plugin.yaml]: Currently only one robot is supported. Multi-Robot support will be added to one of the future releases: https://github.com/thengsty/plugins/projects/1

File: neato/plugin.yaml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    # Global plugin attributes
+    type: interface                   # plugin type (gateway, interface, protocol, system, web)
+    description:
+        de: 'Plugin zur Anbindung von einem Neato Staubsauger Robotor an SmartHomeNG'
+        en: 'Plugin to connect your Neato Vacuum Robot to SmartHomeNG'
+    maintainer: thengsty
+#    tester:                         # Thomas Hengsberg
+    state: develop                  # change to ready when done with development
+#    keywords: iot xyz
+    documentation: https://github.com/smarthomeNG/smarthome/wiki/CLI-Plugin        # url of documentation (wiki) page
```
[the remark this answers is not in the document]
Comment (thengsty) [reply neato/plugin.yaml]: fixed

File: neato/__init__.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+
+class Neato(SmartPlugin):
+    PLUGIN_VERSION = '1.6.0'
+    robot = 'None'
+    _items = []
+
+    def __init__(self, sh, *args, **kwargs):
+        from bin.smarthome import VERSION
+        if '.'.join(VERSION.split('.', 2)[:2]) <= '1.5':
```
[the remark this answers is not in the document]
Comment (thengsty) [reply neato/__init__.py]: statement removed

File: neato/locale.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# translations for the web interface
+plugin_translations:
+    # Translations for the plugin specially for the web interface
+    'Wert 2':         {'de': '=', 'en': 'Value 2'}
+    'Wert 4':         {'de': '=', 'en': 'Value 4'}
+
+    # Alternative format for translations of longer texts:
+    'Hier kommt der Inhalt des Webinterfaces hin.':
```
[the remark this answers is not in the document]
Comment (thengsty) [reply neato/locale.yaml]: webinterface is not used. removed from source

**Pull request merged**